assets/tests/hooks/usePatternsByIdForRoute.test.ts

```diff
@@ -45,75 +49,76 @@ describe("usePatternsByIdForRoute", () => {
       return usePatternsByIdForRoute(null)
     })
     rerender()
-    expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(0)
+    await waitFor(() => expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(0))
   })
 
-  test("doesn't refetch routes on every render", () => {
-    const mockFetchRoutePatterns: jest.Mock =
-      Api.fetchRoutePatterns as jest.Mock
-    mockFetchRoutePatterns.mockReturnValueOnce(
-      instantPromise([routePatternFactory.buildList(2)])
-    )
-    const { rerender } = renderHook(() => {
-      return usePatternsByIdForRoute("66")
+  test("doesn't refetch routes on every render", async () => {
+    jest
+      .mocked(Api.fetchRoutePatterns)
+      .mockResolvedValueOnce(routePatternFactory.buildList(2))
+
+    const { rerender, result } = renderHook(usePatternsByIdForRoute, {
+      initialProps: "66",
     })
-    rerender()
-    expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(1)
+
+    rerender("66")
+
+    await waitFor(() => expect(result.current).not.toBeNull())
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenCalledTimes(1)
   })
 
-  test("refetches when routeId changes", () => {
-    const mockFetchRoutePatterns: jest.Mock =
-      Api.fetchRoutePatterns as jest.Mock
-    mockFetchRoutePatterns.mockReturnValueOnce(
-      instantPromise([routePatternFactory.buildList(2)])
-    )
-    const { rerender } = renderHook(
-      (id) => {
-        return usePatternsByIdForRoute(id)
-      },
-      { initialProps: "66" }
-    )
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("66")
+  test("refetches when routeId changes", async () => {
+    const { rerender } = renderHook(usePatternsByIdForRoute, {
+      initialProps: "66",
+    })
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("66")
+
     rerender("39")
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("39")
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("39")
+    // await waitFor(
+    //   () =>
```

Comment:
🧟

Reply:
Thanks for catching this! Guess I didn't read over my diff enough before setting it for review, done in https://github.com/mbta/skate/compare/2b4f39595fd199a9345d77445cca4fe0106bcc30..1353e383a98b19ba4bd9ffb65da5fe0a9aa7f18d